app/components/move_message.py

```diff
@@ -1,25 +1,231 @@
+    "attachments exceed the limit of 64 MiB:\n"
+    "{offenders}\n"
+    "Please try again without those attachments, or press *Skip* to continue "
+    "without the offending attachments."
+)
+ALL_ATTACHMENTS_TOO_LARGE = (
+    "⚠️ All of your attachments are too large! The limit for the "  # test: allow-vs16
+    "size of attachments is 64 MiB. Please try again with smaller attachments."
+)
+NO_ATTACHMENTS_LEFT = (
+    "Every attachment was selected, but your message only contains "
+    "attachments, which would make the message empty. Would you like to "
+    "delete your message instead?"
+)
+ATTACHMENTS_ONLY = (
+    "Your new message only contains attachments! Doing so will result in your "
+    "message's previous content being removed. Is that what you meant to do?"
+)
+EDITING_TIMEOUT_ALMOST_REACHED = (
+    "{}, are you still editing this message? This thread will be deleted "
+    "{in_five_minutes} if it remains inactive."
+)
+UPLOADING = "⌛ Uploading attachments (this may take some time)…"
+
+
+@dataclass
+class ThreadState:
+    moved_message: MovedMessage
+    split_subtext: SplitSubtext
+    last_update: dt.datetime
+
+
+# A dictionary mapping edit thread IDs to their state. An ID is used instead of
+# the thread itself because there is a thread available already at all places
+# this is used, and because it isn't obvious how threads are hashed.
+edit_threads: dict[int, ThreadState] = {}
+# A dictionary mapping edit thread creator IDs to the ID of the thread they are
+# editing in. IDs are used for the same reason as above.
+edit_thread_creators: dict[int, int] = {}
+
+
+async def _apply_edit_from_thread(
+    moved_message: MovedMessage, message: discord.Message, new_content: str
+) -> None:
+    channel = moved_message.channel
+    # Suppress NotFound in case the user attempts to commit an edit to
+    # a message that was deleted in the meantime.
+    with suppress(discord.NotFound):
+        await moved_message.edit(
+            content=new_content,
+            attachments=[
+                # It is possible that our message to edit is stale if the user
+                # removed an attachment from the message after they started
+                # editing it. Thus, re-fetch the message before grabbing its
+                # attachments to avoid an HTTPException from missing
+                # attachments. It is not stored to a variable beforehand
+                # because fetch_message() returns a Message which can't be used
+                # to edit a webhook message, so we use the old MovedMessage to
+                # perform the edit itself.
+                *(await channel.fetch_message(moved_message.id)).attachments,
+                *(await MessageData.scrape(message)).files,
+            ],
+            allowed_mentions=discord.AllowedMentions.none(),
+        )
+
+
+async def _remove_edit_thread(
+    thread: discord.Thread,
+    author: discord.User | discord.Member,
```

Comment:
Use `app.utils.Account` instead. Also applies to `_remove_edit_thread_after_timeout`.